src/test/java/seedu/address/logic/commands/UpdatePhotoCommandTest.java

```diff
@@ -0,0 +1,173 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import java.nio.file.Paths;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.parser.AddressBookParser;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.ModelManager;
+
+public class UpdatePhotoCommandTest {
+
+    private final String safePath = Paths.get("").toAbsolutePath()
+            + "/src/main/resources/images/test_photo.png";
+
+    @Test
+    public void execute_noIndexCommand_exceptionThrown() {
+        AddressBookParser parser = new AddressBookParser();
+        try {
+            Command command = parser.parseCommand("updatephoto path/images/default_photo.png");
+            assertTrue(command instanceof UpdatePhotoCommand);
+            UpdatePhotoCommand updatePhotoCommand = (UpdatePhotoCommand) command;
+            updatePhotoCommand.execute(new ModelManager());
+        } catch (ParseException e) {
+            assertEquals(e.getMessage(),
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, UpdatePhotoCommand.MESSAGE_USAGE));
+        } catch (CommandException e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void execute_noPathCommand_exceptionThrown() {
+        AddressBookParser parser = new AddressBookParser();
+        try {
+            Command aCommand = parser.parseCommand(
+                    "add n/John Doe p/98765432 e/johndoe@gmail.com a/John street, block 123, #01-01 t/friend"
+            );
+            Command upCommand = parser.parseCommand("updatephoto 1");
+            assertTrue(aCommand instanceof AddCommand);
+            assertTrue(upCommand instanceof UpdatePhotoCommand);
+            AddCommand addCommand = (AddCommand) aCommand;
+            UpdatePhotoCommand updatePhotoCommand = (UpdatePhotoCommand) upCommand;
+            ModelManager temp = new ModelManager();
+            addCommand.execute(temp);
+            updatePhotoCommand.execute(temp);
+        } catch (ParseException e) {
+            assertEquals(e.getMessage(),
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, UpdatePhotoCommand.MESSAGE_USAGE));
+        } catch (CommandException e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void execute_invalidIndexInEmptyAddressBook_exceptionThrown() {
```

Comment:
perhaps the idea of `InEmptyAddressBook` is not really necessary? as i look at your subsequent test cases, looks like you are trying to apply [test case boundary analysis](https://nus-cs2103-ay2324s1.github.io/website/schedule/week10/topics.html#w10-5-test-cases-boundary-value-analysis) which i believe that making an address book empty isn't really necessary. Just a suggestion